src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/StraxCoinviewRule.cs

```diff
@@ -106,5 +117,32 @@ protected override void AllowSpend(TxOut prevOut, Transaction tx)
 
             // Otherwise allow the spend (do nothing).
         }
+
+        /// <inheritdoc />
+        public override async Task RunAsync(RuleContext context)
+        {
+            if (this.logic != null && context.ValidationContext.BlockToValidate.Header is PosBlockHeader posHeader && posHeader.HasSmartContractFields)
+                await this.logic.RunAsync(base.RunAsync, context).ConfigureAwait(false);
+            else
+                await base.RunAsync(context).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        protected override bool CheckInput(Transaction tx, int inputIndexCopy, TxOut txout, PrecomputedTransactionData txData, TxIn input, DeploymentFlags flags)
+        {
+            if (this.logic != null)
```

Comment:
What is the distinction here? I.e. why would `logic` be null (or in what circumstances)?

Reply:
Mostly for tests, but I've mocked it now in the latest PR, so this can change.